### ISSUE TITLE
AnimationPlayer 3D Transfrom Track Rotation ask for normalized tranform for rotation

### ISSUE THREAD

**Issue by danilw** [godotengine/godot-proposals#738]
**Describe the project you are working on:**
3D animated character

**Describe the problem or limitation you are having in your project:**
Creating new *3D Transfrom Track* in Godot AnimationPlayer, setting rotation is **not usable** for hand-editing, when it presented in GUI. 

Rotation ask for normalized Quat, it make impossible to edit it in GUI.
if set not normalized Quat (as for example, someone may expect it rotation by Rad-s) result it error
```
 The end quaternion must be normalized.
 The start quaternion must be normalized.
```

from [docs class-animation](https://docs.godotengine.org/en/stable/classes/class_animation.html#class-animation)
`int transform_track_insert_key ( int track_idx, float time, Vector3 location, Quat rotation, Vector3 scale )`

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
I ask to make in animation *3D Transfrom Track*-editing **GUI** Rotation by angles or rad, to make it useful for hand-editing.

**Comment by Yosimitso** [comment]
this issue still exists in 3.4.4

for the moment the solution is to add the property track rotation_degrees instead of a 3d transform track

**Comment by Calinou** [comment]
This is resolved in 4.0 thanks to the new dedicated 3D position/rotation/scale tracks, but this can't be backported to 3.x for compatibility reasons.

**Comment by rcjoel2019** [comment]
So what i have to do in order to rotate a bone? They don't have the rotation_degrees property. It can't be backported to 3.x, but a change to GUI with euler angles that translate to quaternions in the backend would be nice, since 4.0 may take some time to a stable release

**Comment by Rulend** [comment]
Bones rotations are still quaternions in 4.0 beta 6, making it not possible to just adjust one axis at a time since the w component isn't calculated automatically. I'm guessing the non-normalized quaterntions are also the reason for no interpolation happening between bone animations created inside the animation editor.

**Comment by federicoAntosiano** [comment]
Same issue on Godot 4.1.1.
I tried everything but nothing happened. The only solution I found is doing it via GDScript, but I think that this issue must be resolved...

**Comment by MeanBeanMachine2024** [comment]
Tried it in 4.1.2 and 4.2 beta 2 and it's still happening. that's actually a big problem.
that makes animating bones in-engine basically impossible.

**Comment by MeanBeanMachine2024** [comment]
Okay I may have found a fix. apparently if you animate with the temporary euler instead on the normal rotations the error doesn't go off anymore and the animation playback works fine

I looked it up and apparently a "normalized quaternion" just uses X Y Z, while the normal rotation tools use XYZ and W, which I am guessing the animation player doesn't like for some reason

**Comment by MeanBeanMachine2024** [comment]
Okay found a much better, more reliable fix! apparently if you just use edit mode to rotate bones it will work fine
<img width="209" alt="Screenshot 2023-10-22 172100" src="https://github.com/godotengine/godot-proposals/assets/142996610/13689e80-f866-4cc9-b1c7-69f46f7517d5">

**Comment by GeekzGamingz** [comment]
Also make sure you have at least one frame in each animation or you'll get the same error.

**Comment by luiscesjr** [comment]
I also have this quaternion issue, on 4.2.1 as well. And none of the tricks or workarounds really work for me. I have at least on frame on each animation, use edit mode for rotations, etc. Still broken

![image](https://github.com/godotengine/godot-proposals/assets/7636427/f67c7139-e187-4ad5-a245-c974e2026aa1)

**Comment by mortcodesweb** [comment]
Any work arounds found for 4.2+?

**Comment by danilw** [comment]
To both messages above:

> Any work arounds found for 4.2+?

Source of this error - not normalized quaternion.
I saw this error in some 3d-old-assets models, but fix to it was just - open in Blender and export again.

Second reason - when you do some "animation-rotation" in code and not normalize your transforms - so fix it.

Also as "junk fix" that no one should use - make gdscript and load all animation keys at startup and normalize them and update back to model. But just reexport model in Blender if this is the case.

Putting normalize on every transform from Godot side - will be huge overhead.

My original message was about - having useful UI to edit transform and not typing "normalized rotation" that is impossible to do in UI.

**Comment by Rouli-M** [comment]
Got this error with a simple animation made directly in Godot 4.5.1. Fixed it by selecting the scale/position properties of the object I wanted to animate directly via **+ > property track** instead of **+ > 3D Scale/Position Track**. No idea what the difference between those to are tbh, still a newbie

**Comment by Calinou** [comment]
> Got this error with a simple animation made directly in Godot 4.5.1. Fixed it by selecting the scale/position properties of the object I wanted to animate directly via **+ > property track** instead of **+ > 3D Scale/Position Track**. No idea what the difference between those to are tbh, still a newbie

The dedicated track types are recommended for most use cases, as they are more optimized and support compression on import (and possibly streaming in the future). However, the downside is that they are limited to linear interpolation (e.g. cubic interpolation is not supported). You also can't specify individual X/Y/Z coordinates to animate using `:x`, `:y` and `:z` as a suffix to the property path.

In short, you should use the 3D translation/rotation/scale tracks for basic use cases or when you have large amounts of keyframes (imported 3D scenes automatically do this for this reason). For more advanced use cases where flexibility is favored over performance, keep using property tracks.

**Comment by Rouli-M** [comment]
> The dedicated track types are recommended for most use cases, as they are more optimized and support compression on import (and possibly streaming in the future). However, the downside is that they are limited to linear interpolation (e.g. cubic interpolation is not supported). You also can't specify individual X/Y/Z coordinates to animate using `:x`, `:y` and `:z` as a suffix to the property path.
> 
> In short, you should use the 3D translation/rotation/scale tracks for basic use cases or when you have large amounts of keyframes (imported 3D scenes automatically do this for this reason). For more advanced use cases where flexibility is favored over performance, keep using property tracks.

Unless I'm mistaken (as I have only a surface understanding of all this), it sounds like the property tracks are for a more niche use, as they should be used when you need to optimize **and** your animations only have linear movement. I'd argue most animations don't use purely linear interpolation in real scenarios (I personally use Bézier most frequently, then constant, then linear). At least I think this little explanation should be somewhere in the interface (maybe it is present and I've missed it)

**Comment by Calinou** [comment]
> I'd argue most animations don't use purely linear interpolation in real scenarios (I personally use Bézier most frequently, then constant, then linear).

Note that animations imported from 3D scenes are baked at 30 FPS by default, so even if the source file used Bézier or cubic interpolation, it'll be baked to a series of linearly interpolated points for efficiency reasons.
